cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -444,17 +444,18 @@ func StartControllers(controllers map[string]InitFunc, s *options.CMServer, root
 		glog.Infof("Started %q", controllerName)
 	}
 
-	// all the remaning plugins want this cloud variable
+	// all the remaining plugins want this cloud variable
 	cloud, err := cloudprovider.InitCloudProvider(s.CloudProvider, s.CloudConfigFile)
 	if err != nil {
 		return fmt.Errorf("cloud provider could not be initialized: %v", err)
 	}
 
+	_, clusterCIDR, err := net.ParseCIDR(s.ClusterCIDR)
+	if err != nil {
+		glog.Warningf("Unsuccessful parsing of cluster CIDR %v: %v", s.ClusterCIDR, err)
+	}
+
```

Comment:
I think I have a preference not to do this, also. Each of these if blocks should logically be its own function, and moving this up here makes them less self contained. And given that this happens once on startup, and that CIDR parsing isn't particularly computationally intensive, this isn't really an efficiency win, either.